pandas/core/series.py

```diff
@@ -1335,8 +1336,8 @@ def idxmax(self, axis=None, skipna=True, *args, **kwargs):
         return self.index[i]
 
     # ndarray compat
-    argmin = idxmin
-    argmax = idxmax
+    argmin = deprecate('argmin', idxmin)
+    argmax = deprecate('argmax', idxmax)
```

Comment:
What about simply extending the functionality of the `deprecate` function to be able to take additional text to display in the warning? That way if we have future functions that have similar nuances related to their deprecation we will be able to display them more easily.